app/controllers/users/piv_cac_authentication_setup_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Users
+  class PivCacAuthenticationSetupController < ApplicationController
+    include UserAuthenticator
+
+    before_action :confirm_two_factor_authenticated
+    before_action :authorize_piv_cac_setup, only: :new
+    before_action :authorize_piv_cac_disable, only: :disable
+
+    def new
+      analytics.track_event(Analytics::USER_REGISTRATION_PIV_CAC_SETUP_VISIT)
+      if params.key?(:token)
```

Comment:
What do you think of using strong parameters, and then putting this check in the Form object such that if the token param is not present, it will return `false` for `result.success?`.

Reply:
The problem is that we have a particular UX flow that we're working with. The return from the identity-pki service is a redirect that has the token. We don't want to introduce yet another interaction on the user's part to make the change after they've already asked us to associate their piv/cac with their account.

We could split this into two endpoints so the redirect back from identity-pki isn't to the referring URL but to a different one. I'll need to think about that.